src/library/scala/collection/convert/AsJavaConverters.scala

```diff
@@ -28,27 +32,28 @@ trait AsJavaConverters {
    * using it via the Java interface will be visible via the Scala interface and vice versa.
    *
    * If the Scala `Iterator` was previously obtained from an implicit or explicit call of
-   * `[[JavaConverters.asScalaIterator]](java.util.Iterator)` then the original Java `Iterator` will
-   * be returned.
+   * `asScala` then the original Java `Iterator` will be returned.
    *
    * @param i The Scala `Iterator` to be converted.
    * @return  A Java `Iterator` view of the argument.
    */
-  def asJavaIterator[A](i: Iterator[A]): ju.Iterator[A] = i match {
+  def asJava[A](i: Iterator[A]): ju.Iterator[A] = i match {
     case null                       => null
     case JIteratorWrapper(wrapped)  => wrapped.asInstanceOf[ju.Iterator[A]]
     case _                          => IteratorWrapper(i)
   }
 
+  @deprecated("Use `asJava` instead", "2.13.0")
+  def asJavaIterator[A](i: Iterator[A]): ju.Iterator[A] = asJava(i)
```

Comment:
Wouldn't it be better, since the `jdk` thing is new, to put all of these deprecated methods in their own trait, and have only the old thing inherit them all?  (At that point, I'm not entirely sure they even need to be deprecated, since the trait would be deprecated, and the thing that inherits the trait would be also.)

This way the new API starts out clean.

Reply:
Good idea, thanks!